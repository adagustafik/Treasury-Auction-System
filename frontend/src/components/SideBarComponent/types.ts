export interface IProps {
  page: string;
}
