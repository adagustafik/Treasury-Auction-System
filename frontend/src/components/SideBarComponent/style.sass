@use "../../App"

.side-bar-container
    position: relative
    width:200px
    height: 88.5vh
    border: 2px solid black
    float: left
    padding: 0
    margin: 0
    text-transform: uppercase
    text-align: center

.sidebar-link
    background-color: App.$buttonColor
    color: #ffff

.logout
    position: absolute
    width:100%
    bottom: 0
    left: 0
    background-color: black
    color: #ffff
    border: 0
    border-radius: 0

.side-bar-container button
    width: 100%
    color: black
    background: none
    border: 0

.clicked
    background-color: #0d6efd

.logout-box button
    background-color: App.$buttonColor    
