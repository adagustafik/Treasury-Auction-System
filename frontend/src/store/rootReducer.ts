import { combineReducers } from "@reduxjs/toolkit";
import { auctionsReducer } from "./slices/auctions";
import { userReducer } from "./slices/user";
import { bidsReducer } from "./slices/bids";
import { purchasesReducer } from "./slices/purchases";

export const rootReducer = combineReducers({
  userReducer,
  auctionsReducer,
  purchasesReducer,
  bidsReducer,
});
