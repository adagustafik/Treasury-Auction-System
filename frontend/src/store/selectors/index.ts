export * from "./selectUser";
export * from "./selectAuctions";
export * from "./selectBids";
export * from "./selectPurchases";
