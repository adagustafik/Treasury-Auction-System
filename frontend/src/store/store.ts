import {
  configureStore,
  ThunkAction,
  Action,
  AnyAction,
  ThunkDispatch,
} from "@reduxjs/toolkit";
import thunkMiddleware from "redux-thunk";
import { rootReducer } from "./rootReducer";

export const store = configureStore({
  reducer: rootReducer,
  middleware: [thunkMiddleware],
});

export type AppDispatch = typeof store.dispatch &
  ThunkDispatch<RootState, null, AnyAction>;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
