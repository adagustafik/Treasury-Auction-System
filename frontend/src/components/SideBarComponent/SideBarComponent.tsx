import React, { useEffect } from "react";
import { Button } from "react-bootstrap";
import toast from "react-hot-toast";
import { Link, useNavigate } from "react-router-dom";
import { fetchLogout } from "../../api";
import { fetchResolution } from "../../api/getResolutionRequest";
import { useAppDispatch, useAppSelector } from "../../store/hooks";
import { selectUser } from "../../store/selectors";
import { resetUserDataAction } from "../../store/slices/user";
import "./style.sass";
import { IProps } from "./types";

export function SideBarComponent({ page }: IProps) {
  useEffect(() => {
    const actualLink = document.querySelector(`#${page}`);
    actualLink?.classList.add("clicked");
  });

  const { admin } = useAppSelector(selectUser);
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  const handleLResolution = () => {
    fetchResolution()
      .then((res) => {
        if (res.success) {
          navigate("/dashboard");
          toast(res.success);
        }
      })
      .catch((err) => {
        toast(err.message);
      });
  };

  const handleLogout = () => {
    fetchLogout()
      .then((res) => {
        if (res.success) {
          navigate("/");
          toast(res.success);
          dispatch(resetUserDataAction);
        }
      })
      .catch((err) => {
        toast(err.message);
      });
  };

  return (
    <div className="side-bar-container">
      <div className="d-grid gap-2">
        <Link id="dashboard" className="sidebar-link" to="/dashboard">
          <Button className="sidebar-btn" size="lg">
            Dashboard
          </Button>
        </Link>
        <Link id="profile" className="sidebar-link" to="/profile">
          <Button className="sidebar-btn" size="lg">
            Profile
          </Button>
        </Link>
        <Link id="auctions" className="sidebar-link" to="/auctions">
          <Button className="sidebar-btn" size="lg">
            Auctions
          </Button>
        </Link>
        {admin === false && (
          <Link id="bids" className="sidebar-link" to="/bids">
            <Button className="sidebar-btn" size="lg">
              Bids
            </Button>
          </Link>
        )}
        {admin === true && (
          <Link
            id="create-auction"
            className="sidebar-link"
            to="/create-auction"
          >
            <Button className="sidebar-btn" size="lg">
              Create Auction
            </Button>
          </Link>
        )}
        {admin === true && (
          <div className="sidebar-link">
            <Button
              className="sidebar-btn"
              size="lg"
              onClick={handleLResolution}
            >
              Resolve auctions
            </Button>
          </div>
        )}
        <div className="logout-box">
          <Button className="logout" size="lg" onClick={handleLogout}>
            Logout
          </Button>
        </div>
      </div>
    </div>
  );
}
