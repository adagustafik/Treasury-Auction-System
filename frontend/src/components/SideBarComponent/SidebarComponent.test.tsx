import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { store } from "../../store/store";
import { SideBarComponent } from "./SideBarComponent";

it("renders correctly the SideBar component with prop dashboard", () => {
  const sideBar = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <SideBarComponent page="dashboard" />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(sideBar).toMatchSnapshot();
});
