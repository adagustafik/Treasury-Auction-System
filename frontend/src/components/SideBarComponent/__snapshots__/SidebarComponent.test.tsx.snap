// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the SideBar component with prop dashboard 1`] = `
<div
  className="side-bar-container"
>
  <div
    className="d-grid gap-2"
  >
    <a
      className="sidebar-link"
      href="/dashboard"
      id="dashboard"
      onClick={[Function]}
    >
      <button
        className="sidebar-btn btn btn-primary btn-lg"
        disabled={false}
        type="button"
      >
        Dashboard
      </button>
    </a>
    <a
      className="sidebar-link"
      href="/profile"
      id="profile"
      onClick={[Function]}
    >
      <button
        className="sidebar-btn btn btn-primary btn-lg"
        disabled={false}
        type="button"
      >
        Profile
      </button>
    </a>
    <a
      className="sidebar-link"
      href="/auctions"
      id="auctions"
      onClick={[Function]}
    >
      <button
        className="sidebar-btn btn btn-primary btn-lg"
        disabled={false}
        type="button"
      >
        Auctions
      </button>
    </a>
    <a
      className="sidebar-link"
      href="/bids"
      id="bids"
      onClick={[Function]}
    >
      <button
        className="sidebar-btn btn btn-primary btn-lg"
        disabled={false}
        type="button"
      >
        Bids
      </button>
    </a>
    <div
      className="logout-box"
    >
      <button
        className="logout btn btn-primary btn-lg"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
